impacket/dcerpc/v5/dhcpm.py

```diff
@@ -145,6 +145,9 @@ class DHCP_CLIENT_INFO_VQ(NDRSTRUCT):
     )
 
 class DHCP_CLIENT_SEARCH_UNION(NDRUNION):
+    commonHdr = (
+        ('tag', DHCP_SEARCH_INFO_TYPE),
+    )
```

Comment:
These three lines define the size of the Union tag (defined [here](http://pubs.opengroup.org/onlinepubs/9629399/chap14.htm#tagcjh_19_03_08)]. By default it is [two bytes](https://github.com/CoreSecurity/impacket/blob/master/impacket/dcerpc/v5/ndr.py#L1204) long for NDR32 but on some interfaces it is defined as a 4 bytes long. 
So you should either set a tag type as `NDRSHORT`, `DWORD` or similar. 
Defining the tag as `DHCP_SEARCH_INFO_TYPE` just doesn't seem right, since that's a `NDRENUM` type.